### PR TITLE
[GHI #70] Sidebar

### PR DESCRIPTION
**Pull request by Jeremy-Walton**
## Task

#70 

## What Changed

* [X] Add more breakpoints to prevent leaving a 1px dead space.
* [X] Add layout styles along with documentation
* [X] Refactor the rail to be combined with a new sidebar drawer

## Sanity Check

* ~~Have you updated any usage of changed tokens?~~
* [X] Have you updated the docs with any component changes?
* [X] Do you need to update the package version?
* [ ] Add link from sidebar docs to layout to show how to place it in the context of your project

## Screenshots

Screen Shot 2022-10-12 at 6 37 22 PM
Screen Shot 2022-10-12 at 6 37 28 PM
Screen Shot 2022-10-12 at 6 37 52 PM

### CODE REVIEW

File: src/core/tokens/base_tokens.scss [hunk 1]
```diff
@@ -159,11 +160,19 @@
 // https://drafts.csswg.org/css-env-1/
 // https://bholmes.dev/blog/alternative-to-css-variable-media-queries/
 
-$breakpoint-x-small: 512px; // phone
+// The below values are to ensure media queries that define a boundary
+// don't leavea 1px dead space
```
Comment (Jeremy-Walton) [inline src/core/tokens/base_tokens.scss:164]: Typo

File: src/core/layout.scss [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    min-height: 1px;
+    margin-top: -1px;
+  }
+}
+
+.page__header {
+  flex-grow: 0;
+  flex-shrink: 0;
+
+  position: sticky;
```
Comment (Jeremy-Walton) [inline src/core/layout.scss:28]: Should add a note about the sticky header and footer in the docs

File: src/stories/Recipes/Layout/Layout.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+<div class='app-wrapper' style="height: 40rem;"> <!-- This class should be on body. Height is for demo purposed -->
+  <!-- Flash message here -->
+  <!-- Confirm here -->
+  <!-- Modal here -->
+  <!-- Panel here -->
+  ${sidebar.innerHTML}
+  <div class='app__main'>
+    <div class='page__header background-primary-plus-3'>Header</div>
+    <div class='page__content'>
+      <p>Some content</p>
```
Comment (Jeremy-Walton) [inline src/stories/Recipes/Layout/Layout.js:40]: Map or loop to prevent the duplication

**Review by jdmcleod — APPROVED** [review]

File: src/components/sidebar.scss [hunk 4]
```diff
@@ -1,97 +1,254 @@
...
+      white-space: nowrap;
+      overflow: hidden;
+      text-overflow: ellipsis;
+    }
   }
 
-  &:last-of-type {
-    margin-bottom: 0;
+  // Modifiers
+  &.sidebar--drawer:not(.sidebar--collapsible) {
```
Comment (Jeremy-Walton) [inline src/components/sidebar.scss:197]: Instead of collapsible, this should be "responsive"

File: src/stories/Sidebar/Sidebar.mdx [hunk 5]
```diff
@@ -0,0 +1,194 @@
+import { Primary, ArgsTable, Canvas, Story } from "@storybook/addon-docs";
```
Comment (Jeremy-Walton) [inline src/stories/Sidebar/Sidebar.mdx:1]: Let's add an example of a highly customized sidebar that can be used as a template for an app to reference

File: src/stories/Recipes/Layout/Layout.mdx [hunk 6]
```diff
@@ -0,0 +1,75 @@
+import { Primary, ArgsTable, Canvas, Story } from "@storybook/addon-docs";
```
Comment (Jeremy-Walton) [inline src/stories/Recipes/Layout/Layout.mdx:1]: Let's add a new example in here and the playground so we can show a layout with the sidebar on the right

**Review by scriswell — APPROVED** [review]
@Jeremy-Walton Approved with our discussed updates.

**Pull request merged**